manifests/suse.pp

```diff
@@ -56,14 +67,15 @@
   }
 
   zypprepo { 'datadog':
-    baseurl      => $baseurl,
-    enabled      => 1,
-    autorefresh  => 1,
-    name         => 'datadog',
-    gpgcheck     => 1,
+    baseurl       => $baseurl,
+    enabled       => 1,
+    autorefresh   => 1,
+    name          => 'datadog',
+    gpgcheck      => 1,
     # zypper on SUSE < 15 only understands a single gpgkey value
-    gpgkey       => (Float($::operatingsystemmajrelease) >= 15.0) ? { true => join($gpgkeys, "\n       "), default => $current_key },
-    keeppackages => 1,
+    gpgkey        => (Float($::operatingsystemmajrelease) >= 15.0) ? { true => join($gpgkeys, "\n       "), default => $current_key },
+    repo_gpgcheck => $repo_gpgcheck,
```

Comment:
I am a bit confused, I added this on voxpupuli/puppet-zypprepo#48 and I thought it was only available since version 4.0.0 of the zypprepo dependency 🤔 Does it work?

Reply:
The changelog says it's supported since 3.1.0: https://forge.puppet.com/modules/puppet/zypprepo/3.1.0/changelog#v310-2020-12-12, which is the minimal version that we depend on, according to `manifest.json`, so I think we should be fine. Correct me if I'm wrong :)

Reply:
The changelog is wrong x) My PR was accidentally reverted, see https://github.com/voxpupuli/puppet-zypprepo/pull/52#discussion_r553912647. It was added back on voxpupuli/puppet-zypprepo@399244480c270d611167a2f56846672c75b4ea3c which is only available on 4.0.0.

Reply:
Oh, I see. Well, I'm wondering why this is not making the unittests fail - it seems like we are actually installing 4.0.0 when testing?

https://app.circleci.com/pipelines/github/DataDog/puppet-datadog-agent/429/workflows/792ba89c-4e9b-41c0-87d9-f89064b4e9ee/jobs/12379 contains:

```
└── puppet-zypprepo (v4.0.0)
```

I'll need to investigate.

Reply:
Yeah, it was caused by unpinned version of zypprepo. I pinned it for tests now and I also commented out `repo_gpgcheck` for `suse` - the comment explains why this doesn't actually hurt us too much.